cmd/kubeadm/app/cmd/join.go

```diff
@@ -443,12 +451,25 @@ func newJoinData(cmd *cobra.Command, args []string, opt *joinOptions, out io.Wri
 		}
 	}
 
+	// if dry running, creates a temporary folder to save kubeadm generated files
+	dryRunDir := ""
+	if opt.dryRun {
+		// the KUBEADM_JOIN_DRYRUN_DIR environment variable allows overriding the dry-run temporary
+		// directory from the command line. This makes it possible to run "kubeadm join" integration
+		// tests without root.
+		if dryRunDir, err = kubeadmconstants.CreateTempDirForKubeadm(os.Getenv("KUBEADM_JOIN_DRYRUN_DIR"), "kubeadm-join-dryrun"); err != nil {
+			return nil, errors.Wrap(err, "couldn't create a temporary directory")
+		}
```

Comment:
in reality KUBEADM_JOIN_DRYRUN_DIR is not used.
i think we can leave this to be:
```suggestion
		if dryRunDir, err = kubeadmconstants.CreateTempDirForKubeadm("", "kubeadm-join-dryrun"); err != nil {
			return nil, errors.Wrap(err, "couldn't create a temporary directory")
		}
```

Reply:
ok, fixed